plugins/plugin-chart-concise-card/src/plugin/buildQuery.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { buildQueryContext, QueryFormData } from '@superset-ui/core';
+
+/**
+ * The buildQuery function is used to create an instance of QueryContext that's
+ * sent to the chart data endpoint. In addition to containing information of which
+ * datasource to use, it specifies the type (e.g. full payload, samples, query) and
+ * format (e.g. CSV or JSON) of the result and whether or not to force refresh the data from
+ * the datasource as opposed to using a cached copy of the data, if available.
+ *
+ * More importantly though, QueryContext contains a property `queries`, which is an array of
+ * QueryObjects specifying individual data requests to be made. A QueryObject specifies which
+ * columns, metrics and filters, among others, to use during the query. Usually it will be enough
+ * to specify just one query based on the baseQueryObject, but for some more advanced use cases
+ * it is possible to define post processing operations in the QueryObject, or multiple queries
+ * if a viz needs multiple different result sets.
+ */
+export default function buildQuery(formData: QueryFormData) {
+  const { cols: groupby } = formData;
+
+  return buildQueryContext(formData, baseQueryObject => {
+    const tagQuery = {
+      ...baseQueryObject,
+      columns: ['tag'],
+      metrics: [],
+      filters: [],
+    };
+
+    const categoryQuery = {
+      ...baseQueryObject,
+      columns: ['category'],
+      metrics: [],
+      filters: [],
+    };
```

Comment:
is there any situation where we'd want to have column names other than 'category' and 'tag'?  you might consider requiring that groupby is length 2 and then using the first entry in groupby as your 'category' and the second entry as 'tag' so that the component has more versatility?  i don't know if there's a use case that requires doing so, but you might consider it.

Reply:
I've discussed it with Bogdan and he thinks it's a good idea. So i'll make it that way.

Reply:
I've changed it to be dynamic. Please review.